cpp/cmake_modules/FindClangTools.cmake

```diff
@@ -75,7 +75,7 @@ function(FIND_CLANG_TOOL NAME OUTPUT VERSION_CHECK_PATTERN)
 endfunction()
 
 find_clang_tool(clang-tidy CLANG_TIDY_BIN
-                "LLVM version ${ARROW_LLVM_MAJOR_VERSION}\\.${ARROW_LLVM_MINOR_VERSION}")
+                "LLVM version ${ARROW_LLVM_MAJOR_VERSION}")
```

Comment:
It wasn't able to detect clang tools for me, because I have llvm 7.1. 
The LLVM_VERSION - minor as well - is hardcoded here:

https://github.com/apache/arrow/blob/3f2a33f902983c0d395e0480e8a8df40ed5da29c/cpp/CMakeLists.txt#L91-L99

Reply:
You can probably disable Gandiva in the meantime :-) I suggest opening a JIRA about this issue and reverting this change.

Reply:
Does it make sense to restrict the minor version too for clang tools?

Reply:
I think so, because they may format code differently. In any case, I think this should be a separate JIRA.

Reply:
https://issues.apache.org/jira/browse/ARROW-6514